osf_tests/test_generate_sitemap.py

```diff
@@ -0,0 +1,164 @@
+import os
+
+import pytest
+import mock
+import shutil, tempfile, xml, urlparse
+
+from scripts import generate_sitemap
+from osf_tests import factories
+from website import settings
+
+
+def get_all_sitemap_urls():
+    # Create temporary directory for the sitemaps to be generated
+
+    generate_sitemap.main()
+
+    # Parse the generated XML sitemap file
+    with open(settings.STATIC_FOLDER + '/sitemaps' + '/sitemap_0.xml') as f:
+        tree = xml.etree.ElementTree.parse(f)
+
+    shutil.rmtree(settings.STATIC_FOLDER)
+
+    # Get all the urls in the sitemap
+    # Note: namespace was defined in the XML file, therefore necessary to include in tag
+    namespace = '{http://www.sitemaps.org/schemas/sitemap/0.9}'
+    urls = [element.text for element in tree.iter(namespace + 'loc')]
+
+    return urls
+
+
+@pytest.mark.django_db
+class TestGenerateSitemap:
+
+    @pytest.fixture(autouse=True)
```

Comment:
I don't usually use the autouse fixture, but if it something that needs to be setup for all of the tests in the file it probably doesn't hurt to have that